django_project/realtime/tasks/realtime/test/test_realtime_tasks.py

```diff
@@ -47,4 +63,78 @@ def test_process_ash(self):
             eruption_height=100,
             forecast_duration=3)
 
+        # force synchronous result
+        ash.refresh_from_db()
+        result = AsyncResult(id=ash.task_id, app=realtime_app)
+        actual_value = result.get()
+        ash.refresh_from_db()
+
+        # Check state
+        self.assertEqual(result.state, 'SUCCESS')
+
+        # Check ret_val
+        expected_value = {
+            'hazard_path': '/home/realtime/ashmaps/201702211904+0700_Merapi/'
+                           'ash_fall.tif',
+            'success': True
+        }
+        self.assertEqual(expected_value, actual_value)
+
+        # Check hazard information pushed to db
+        self.assertTrue(ash.hazard_layer_exists)
+
+        result = check_processing_task.delay()
+        result.get()
+
+        ash.refresh_from_db()
+
+        # TODO: Fixme. Somehow the following asserts produces an error
```

Comment:
May be the task is indeed not finished... Perhaps add sleep 1 minute or something.

Reply:
The task already finished at the top, but the result.state changed from SUCCESS to PENDING again if inspected.
If I run it in debug mode, this problem won't happen. I was confused.